mssql/mssql.yml

```diff
@@ -0,0 +1,876 @@
+                  |> filter(fn: (r) => r["_field"] == "Percent_Disk_Read_Time")
+        width: 3
+        xCol: _time
+        xPos: 9
+        yCol: _value
+        yPos: 1
+      - colors:
+          - hex: '#00C9FF'
+            name: laser
+            type: text
+        decimalPlaces: 0
+        height: 1
+        kind: Single_Stat
+        name: Pending
+        queries:
+          - query: |-
+                from(bucket: "mssql")
+                  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
+                  |> filter(fn: (r) => r["_measurement"] == "sqlserver_server_properties")
+                  |> filter(fn: (r) => r["_field"] == "db_recoveryPending")
+        width: 1
+        xPos: 10
+      - colors:
+          - hex: '#00C9FF'
+            name: laser
+            type: text
+        decimalPlaces: 0
+        height: 1
+        kind: Single_Stat
+        name: Suspect
+        queries:
+          - query: |-
+                from(bucket: "mssql")
+                  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
+                  |> filter(fn: (r) => r["_measurement"] == "sqlserver_server_properties")
+                  |> filter(fn: (r) => r["_field"] == "db_suspect")
+        width: 1
+        xPos: 11
+    name: MSSQL
+
+---
+
+apiVersion: influxdata.com/v2alpha1
+kind: Telegraf
+metadata:
+    name: mssql-config
+spec:
+    config: |
+          [[outputs.influxdb_v2]]
+            ## The URLs of the InfluxDB cluster nodes.
+            ##
+            ## Multiple URLs can be specified for a single cluster, only ONE of the
+            ## urls will be written to each interval.
+            ## urls exp: http://127.0.0.1:9999
+            urls = ["$INFLUX_HOST"]
+
+            ## Token for authentication.
+            token = "$INFLUX_TOKEN"
+
+            ## Organization is the name of the organization you wish to write to; must exist.
+            organization = "$INFLUX_ORG"
+
+            ## Destination bucket to write into.
+            bucket = "$INFLUX_BUCKET"
+
+          [agent]
+            ## Default data collection interval for all inputs, can be changed as per collection interval needs
+            interval = "10s"
+
+          # Read metrics from Microsoft SQL Server
+          [[inputs.sqlserver]]
+            ## Specify instances to monitor with a list of connection strings.
+            ## All connection parameters are optional.
+            ## By default, the host is localhost, listening on default port, TCP 1433.
+            ##   for Windows, the user is the currently running AD user (SSO).
+            ##   See https://github.com/denisenkom/go-mssqldb for detailed connection
+            ##   parameters, in particular, tls connections can be created like so:
+            ##   "encrypt=true;certificate=<cert>;hostNameInCertificate=<SqlServer host fqdn>"
+            # servers = [
+            #  "Server=192.168.1.10;Port=1433;User Id=<user>;Password=<pw>;app name=telegraf;log=1;",
```

Comment:
Could you make this an environment variable (the whole string as one variable, or multiple variables for each component) so that it can be run via `telegraf --url` without modification?

Reply:
Michael, about the change request. This is telegraf input "config" for mssql came by default. I do not touch this. In fact, when you run locally telegraf in the same server with mssql, you don't need to specify any host

Reply:
https://github.com/influxdata/telegraf/tree/master/plugins/inputs/sqlserver